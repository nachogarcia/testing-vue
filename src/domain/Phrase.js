const longPhraseThreshold = 15

class Phrase {
  constructor ({ id, categories, text }) {
    this.id = id
    this.categories = categories
    this.text = text
  }

  get isNerdy () {
    return this.categories.includes('nerdy')
  }

  get isLong () {
    return this.text.length > longPhraseThreshold
  }
}

export default Phrase
