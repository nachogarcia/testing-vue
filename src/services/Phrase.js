import axios from 'axios'
import Phrase from '@/domain/Phrase'

const URL = 'https://api.icndb.com/jokes/' //Future environment variable

const parse = data => new Phrase({
  id: data.id,
  categories: data.categories,
  text: data.joke
})

class PhraseService {
  async getRandomPhrases(numberOfPhrases) {
    const response = await axios.get(URL + 'random/' + numberOfPhrases)
    return response.data.value.map( data => parse(data) )
  }

  async getMostImportantPhrase() {
    const response = await axios.get(URL + '503')
    return parse(response.data.value)
  }
}

export default new PhraseService()
