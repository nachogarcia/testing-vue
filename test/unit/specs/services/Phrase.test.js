import phraseService from '@/services/Phrase'
import Phrase from '@/domain/Phrase'

describe('Phrase service', () => {
  it('gets random phrases', async () => {
    const numberOfPhrases = 5

    const phrases = await phraseService.getRandomPhrases(numberOfPhrases)

    expect(phrases).toHaveLength(numberOfPhrases)
    phrases.forEach(phrase => {
      expect(phrase).toBeInstanceOf(Phrase)
      expect(phrase.id).toBeDefined()
      expect(phrase.categories).toBeInstanceOf(Array)
      expect(typeof phrase.text).toBe('string')
    })
  })

  it('gets the most important phrase', async () => {
    const phrase = await phraseService.getMostImportantPhrase()

    expect(phrase).toMatchSnapshot()
  })
})
