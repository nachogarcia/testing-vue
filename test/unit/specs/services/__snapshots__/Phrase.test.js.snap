// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Phrase service gets the most important phrase 1`] = `
Phrase {
  "categories": Array [
    "nerdy",
  ],
  "id": 503,
  "text": "Chuck Norris protocol design method has no status, requests or responses, only commands.",
}
`;
